### PR TITLE
Updates to VeinholeMonsters, Tiberium and SWs + minor stuff

### PR DESCRIPTION
**Pull request by ZivDero**

### CODE REVIEW

File: BuildingClass.h [hunk 1]
```diff
@@ -72,13 +72,18 @@ class NOVTABLE BuildingClass : public TechnoClass
...
-	bool IsAllFogged() const { JMP_THIS(0x457A10); }
+	bool IsAllFogged() const
+		{ JMP_THIS(0x457A10); }
 
 	void SetRallypoint(CellStruct* pTarget, bool bPlayEVA)
 		{ JMP_THIS(0x443860); }
 
+	int GetPipCount()
+		{ JMP_THIS(0x44D700); }
+
```
Comment (Starkku) [inline BuildingClass.h]: This is unnecessary addition. This function is a virtual function and is already available in `TechnoClass` under name `GetPipFillLevel()`.

**Review by Starkku — CHANGES_REQUESTED** [review]
One small thing that should be addressed but other than that looks good to me.

File: BuildingClass.h (hunk 1 above)
Comment (ZivDero) [reply BuildingClass.h]: Done.

**Comment by Metadorius** [comment]
@Starkku have you compared the stuff with our IDB? as in do I need to do that in case you didn't?

File: HouseClass.h [hunk 2]
```diff
@@ -713,6 +713,9 @@ class NOVTABLE HouseClass : public AbstractClass, public IHouse, public IPublicH
 	void AddTracking(TechnoClass* pTechno)
 		{ JMP_THIS(0x4FF700); }
 
+	long double GetWeedStoragePercentage()
```
Comment (Metadorius) [inline HouseClass.h]:
```suggestion
	double WeedFraction()
```
At least that's what it is in my copy of mappings, also not sure what's `long` for

File: ParticleClass.h [hunk 3]
```diff
@@ -60,8 +60,8 @@ class NOVTABLE ParticleClass : public ObjectClass
 	DWORD  unknown_B4;
 	DWORD  unknown_B8;
 	DWORD  unknown_BC;
-	DWORD  unknown_C0;
-	DWORD  unknown_C4;
+	int  VelocityX;
+	int  VelocityY;
 	DWORD  unknown_C8;
```
Comment (Metadorius) [inline ParticleClass.h]: image

File: SmudgeTypeClass.h [hunk 4]
```diff
@@ -35,6 +35,9 @@ class NOVTABLE SmudgeTypeClass : public ObjectTypeClass
 	//SmudgeTypeClass
 	virtual void DrawIt(const Point2D& Point, const RectangleStruct& Rect, int SmudgeData, int Height, const CellStruct& MapCoords) RX;
 
+	static void __fastcall LoadFromIniList(int idxTheatre)
```
Comment (Metadorius) [inline SmudgeTypeClass.h:38]: called `Init` in my IDB

File: VeinholeMonsterClass.h [hunk 5]
```diff
@@ -11,41 +12,41 @@ class VeinholeLogic
...
-	int Count;
-	PointerHeapClass<MapSurfaceData>* Heap;
-	MapSurfaceData* Datas;
-	CDTimerClass Timer;
-	bool* States;
+	int NextFreeNodeIndex;
+	PriorityQueueClass<PriorityQueueClassNode>* Queue;
+	PriorityQueueClassNode* Nodes;
+	CDTimerClass VeinTimer;
+	bool* CellIndexesWithVeins;
```
Comment (Metadorius) [inline VeinholeMonsterClass.h:49]: Seems to be used in multiple places in TibClass at least, see end of the class:

```cpp
struct __declspec(align(8)) TiberiumClass
{
  AbstractTypeClass at;
  int Type;
  int Spread;
  double SpreadPercentage;
  int Growth;
  double GrowthPercentage;
  int Value;
  int Power;
  int Color;
  TypeList Debris;
  int Image;
  int __NumFrames;
  int __NumImages;
  int int_field_EC;
  int SpreadCount;
  TiberiumClass_Data *SpreadHeaps;
  bool *SpreadState;
  TiberiumClass_CellData *SpreadCellData;
  TimerClass __SpreadTimer;
  int GrowthCount;
  TiberiumClass_Data *GrowthHeaps;
  bool *GrowthState;
  TiberiumClass_CellData *GrowthCellData;
  TimerClass __GrowthTimer;
};

struct TiberiumClass_Data
{
  int Size;
  int Capacity;
  TiberiumClass_CellData **Heap;
  int Ptr1;
  int Ptr2;
};

struct TiberiumClass_CellData
{
  CellStruct Cell;
  float Score;
};
```

File: VeinholeMonsterClass.h [hunk 6]
```diff
@@ -128,7 +129,7 @@ class NOVTABLE VeinholeMonsterClass : public ObjectClass
 	}
 
 	//called 687A80
-	static void __fastcall InitVeiGrowhData(bool bAllocate = true)
```
Comment (Metadorius) [inline VeinholeMonsterClass.h:131]: this is the vei

File: VeinholeMonsterClass.h [hunk 7]
```diff
@@ -152,6 +153,18 @@ class NOVTABLE VeinholeMonsterClass : public ObjectClass
 	static void __fastcall DrawAll()
 	{ JMP_STD(0x74D430); }
 
+	static void __fastcall DeleteAll()
+	{ JMP_STD(0x74D760); }
+
+	static void __fastcall DeleteVeinholeGrowthData()
```
Comment (Metadorius) [inline VeinholeMonsterClass.h:159]: called Spread, not Growth in mine

File: VeinholeMonsterClass.h [hunk 8]
```diff
@@ -152,6 +153,18 @@ class NOVTABLE VeinholeMonsterClass : public ObjectClass
...
+	{ JMP_STD(0x74E880); }
+
+	static void __fastcall LoadVeinholeArt(int idxTheatre)
```
Comment (Metadorius) [inline VeinholeMonsterClass.h:162]: called Init in mine

File: VeinholeMonsterClass.h [hunk 9]
```diff
@@ -152,6 +153,18 @@ class NOVTABLE VeinholeMonsterClass : public ObjectClass
...
+	{ JMP_STD(0x74D450); }
+
+	static void __cdecl UpdateAllVeinholes()
```
Comment (Metadorius) [inline VeinholeMonsterClass.h:165]: this is called `Remove_All_Veinhole_Monsters` in mine, possibly an error in IDB?

**Review by Metadorius — COMMENTED** [review]
Commendable work 👍 
Left some stuff to discuss or for reference if needed but otherwise should be mergeable.

File: TiberiumClass.h [hunk 10]
```diff
@@ -5,64 +5,45 @@
...
 	}
 
 	int Count;
-	PointerHeapClass<MapSurfaceData>* Heap;
-	bool* States;
-	MapSurfaceData* Datas;
+	PriorityQueueClass<PriorityQueueClassNode>* Queue;
+	bool* CellIndexesWithTiberium;
+	PriorityQueueClassNode* Nodes;
 	CDTimerClass Timer;
```
Comment (Metadorius) [inline TiberiumClass.h:47]: Wait didn't notice the duplication. Guess it will be good to generalize that in future.

File: HouseClass.h (hunk 2 above)
Comment (ZivDero) [reply HouseClass.h]: Probably nothing.

File: VeinholeMonsterClass.h (hunk 9 above)
Comment (ZivDero) [reply VeinholeMonsterClass.h:165]: Most likely an error in the IDB, yes.

**Review by Metadorius — APPROVED** [review]

**Pull request merged**